src/xform_to_datalog/authorization_logic_test.h

```diff
@@ -0,0 +1,55 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#include "src/xform_to_datalog/authorization_logic.h"
+
+#include <cstdlib>
+#include <filesystem>
+#include <fstream>
+#include <vector>
+
+#include "src/common/testing/gtest.h"
+
+namespace fs = std::filesystem;
+
+namespace raksha::xform_to_datalog {
+
+class AuthorizationLogicTest : public ::testing::Test {
+public:
+  static fs::path GetTestDataDir() {
```

Comment:
Why are these methods `static` and `public`? I think the preferred pattern is to make test fixtures fill in `protected` fields via their default constructor or via `SetUp`, defining all members as protected. The `protected` fields can then be accessed directly from `TEST_F` or `TEST_P` tests.